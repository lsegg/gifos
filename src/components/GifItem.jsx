import { PropTypes } from "prop-types";
import favourite from "../assets/imgs/icon-fav.svg";
import download from "../assets/imgs/icon-download.svg";
import maximize from "../assets/imgs/icon-max.svg";

const GifItem = ({ title, url }) => {
  return (
    <div className="Search-result">
      <img src={url} alt={title} className="Search-result-gif" />
      <div className="Search-result-gif-overlay">
        <span className="Search-result-gif-overlay-wrapper">
          <img
            src={favourite}
            alt="favourite"
            className="Search-result-gif-overlay-wrapper-action"
          />
          <img
            src={download}
            alt="download"
            className="Search-result-gif-overlay-wrapper-action"
          />
          <img
            src={maximize}
            alt="maximize"
            className="Search-result-gif-overlay-wrapper-action"
          />
        </span>
        <p className="Search-result-gif-overlay-title">{title}</p>
      </div>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifItem;
