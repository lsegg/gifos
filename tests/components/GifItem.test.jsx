import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";
describe("GifItem", () => {
  const title = "Title";
  const url = "https://url.com/";
  test("should match the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show the gif with the right url and alt props", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should show the title in the component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
