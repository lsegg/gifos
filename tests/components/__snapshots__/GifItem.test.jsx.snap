// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="https://url.com/"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
